src/main/java/org/opentripplanner/routing/core/RoutingContext.java

```diff
@@ -143,33 +145,16 @@ public RoutingContext(RoutingRequest routingRequest, Graph graph, Vertex from, V
     }
 
     /**
-     * Returns the StreetEdges that overlap between two vertices edge sets.
+     * Returns the StreetEdges that overlap between two vertices' edge sets.
      */
     private Set<StreetEdge> overlappingStreetEdges(Vertex u, Vertex v) {
-        Set<Integer> vIds = new HashSet<Integer>();
-        Set<Integer> uIds = new HashSet<Integer>();
-        for (Edge e : Iterables.concat(v.getIncoming(), v.getOutgoing())) {
-            vIds.add(e.getId());
-        }
-        for (Edge e : Iterables.concat(u.getIncoming(), u.getOutgoing())) {
-            uIds.add(e.getId());
-        }
-        
-        // Intesection of edge IDs between u and v.
-        uIds.retainAll(vIds);
-        Set<Integer> overlappingIds = uIds;
-
-        // Fetch the edges by ID - important so we aren't stuck with temporary edges.
-        Set<StreetEdge> overlap = new HashSet<>();
-        for (Integer id : overlappingIds) {
-            Edge e = graph.getEdgeById(id);
-            if (e == null || !(e instanceof StreetEdge)) {
-                continue;
-            }
-
-            overlap.add((StreetEdge) e);
-        }
-        return overlap;
+        Set<Edge> vEdges = Sets.newHashSet(Iterables.concat(v.getIncoming(), v.getOutgoing()));
+        Set<Edge> uEdges = Sets.newHashSet(Iterables.concat(u.getIncoming(), u.getOutgoing()));
+        return Sets.intersection(uEdges, vEdges).stream()
+                .filter(Objects::nonNull)
+                .filter(e -> e instanceof StreetEdge)
+                .map(e -> (StreetEdge)e)
+                .collect(Collectors.toSet());
     }
 
```

Comment:
I had problems with the above code, and the TestHalfEdges#testRouteToSameEdge failed. I changed the code to this to mak it work:
```
    /**
     * Returns the StreetEdges that overlap between two vertices' edge sets.
     * It does not look at the TemporaryPartialStreetEdge, but the real parent
     * of this edge.
     */
    private Set<StreetEdge> overlappingParentStreetEdges(TemporaryStreetLocation u, TemporaryStreetLocation v) {
        // Fetch the parent edges so we aren't stuck with temporary edges.
        Set<StreetEdge> vEdges = getConnectedParentEdges(v);
        Set<StreetEdge> uEdges = getConnectedParentEdges(u);
        return Sets.intersection(uEdges, vEdges);
    }

    /**
     * Find all parent edges ({@link TemporaryPartialStreetEdge#getParentEdge()}) for
     * {@link TemporaryStreetLocation#getIncoming()} and
     * {@link TemporaryStreetLocation#getIncoming()} edges. Edges of other types are ignored.
     */
    private static Set<StreetEdge> getConnectedParentEdges(TemporaryStreetLocation loc) {
        return Stream.concat(loc.getIncoming().stream(), loc.getOutgoing().stream())
                .filter(it -> it instanceof TemporaryPartialStreetEdge)
                .map(it -> ((TemporaryPartialStreetEdge)it).getParentEdge())
                .collect(Collectors.toSet());
    }
```
Note! I renamed the method from `overlappingStreetEdges` to `overlappingParentStreetEdges`.

Reply:
Oh, thanks for catching that. I didn't catch that comment about "important so we aren't stuck with temporary edges". My understanding was that every edge including temporary ones had a unique ID, not that split edges would have the same ID as their parent edge. But indeed: `public int getId() { return parentEdge.getId(); }`. I also didn't realize this method was supposed to get intersecting _parent_ edges because the Javadoc was incomplete. Your renaming and Javadoc help with this.